src/dataviews/dataview-model-base.js

```diff
@@ -127,6 +130,10 @@ module.exports = Model.extend({
         silent: silent,
         forceFetch: forceFetch
       });
+
+      if (this.get('sync_on_data_change')) {
+        this._newDataAvailable = true;
```

Comment:
That flag is used [here](https://github.com/CartoDB/cartodb.js/pull/1162/files#diff-805364e485d740e26e33839755aab753R161).